cattle/plugins/volmgr/handler.py

```diff
@@ -69,23 +69,30 @@ def _get_volume_uuid(self, volume):
         return volume_uuid
 
     def _is_snapshot_created(self, snapshot):
-        return 'managedSnapshotUUID' in snapshot.data.fields
+        volume_uuid = self._get_volume_uuid(snapshot.volume)
+        return volmgr.snapshotExists(snapshot.uuid, volume_uuid)
 
     def _do_snapshot_create(self, snapshot, progress):
         log.info("Creating snapshot")
         volume_uuid = self._get_volume_uuid(snapshot.volume)
+        snapshot_uuid = snapshot.uuid
         log.info("Creating snapshot for volume %s" % volume_uuid)
-        snapshot_uuid = volmgr.create_snapshot(volume_uuid)
-        snapshot.data.fields['managedSnapshotUUID'] = snapshot_uuid
-        snapshot.data.fields['managedVolumeUUID'] = volume_uuid
+        snapshot_uuid = volmgr.create_snapshot(volume_uuid,
+                                               snapshot_uuid)
         log.info("New snapshot for volume, uuid is %s" % snapshot_uuid)
         return True
 
     def _is_snapshot_backed_up(self, snapshot, storage_pool):
-        return 'backup' in snapshot.data.fields
+        blockstore_uuid = volmgr.blockstore_uuid
+        if 'blockstoreUUID' in storage_pool.data.fields:
```

Comment:
This is a future-proof thing. Currently all the blockstore operations on the same python-agent would goes to the same blockstore(which is default locally). In the future we would introduce blockstore storage pool for this purpose, so it would have blockstore_uuid to specify which one to operate on.